core/domain/user_jobs_one_off.py

```diff
@@ -324,3 +327,50 @@ def map(user_model):
     @staticmethod
     def reduce(key, stringified_values):
         pass
+
+
+class UserRolesMigrationOneOffJob(jobs.BaseMapReduceJobManager):
+    """A one time job to attach a new field (representing role of user) to the
+    UserSettingsModel. The job will load all existing users from the datastore,
+    look for the role to assign them and then store them back thus updating the
+    table schema and adding appropiate roles to users.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_model):
+        admin_usernames = config_domain.Registry.get_config_property(
+            'admin_usernames')
+        moderator_usernames = config_domain.Registry.get_config_property(
+            'moderator_usernames')
+        banned_usernames = config_domain.Registry.get_config_property(
+            'banned_usernames')
+        collection_editors = config_domain.Registry.get_config_property(
+            'collection_editor_whitelist')
+
+        try:
+            if user_model.username in admin_usernames.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_ADMIN)
+            elif user_model.username in moderator_usernames.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_MODERATOR)
+            elif user_model.username in banned_usernames.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_BANNED_USER)
+            elif user_model.username in collection_editors.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_COLLECTION_EDITOR)
+            else:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_EXPLORATION_EDITOR)
+            yield ('success ', 'Role successfully attached.')
```

Comment:
This isn't really useful. I recommend just removing it. In "reduce" you're just going to end up with ('success', ['Role successfully attached.', 'Role successfully attached.', ...]).

I suggested "reduce on sum" earlier -- see https://en.wikipedia.org/wiki/MapReduce#Examples for an example of what this means. But I don't really think it's necessary.